homeassistant/components/brother/config_flow.py

```diff
@@ -19,14 +23,15 @@
 )
 
 
-def host_valid(host):
+def host_valid(host: str) -> bool:
     """Return True if hostname or IP address is valid."""
     try:
         if ipaddress.ip_address(host).version == (4 or 6):
             return True
```

Comment:
To fix codecov (and remove L34). Mypy should be fine as well.
```suggestion
        return ipaddress.ip_address(host).version == (4 or 6):
```

Reply:
Well, actually codecov/patch passes even without it.

Reply:
There is an error with this change:
```
homeassistant/components/brother/config_flow.py:29: error: Returning Any from function declared to return "bool"  [no-any-return]
```

Reply:
I realised this comparison contains bug. `(4 or 6)` is bitwise operation and it equals to `4`. So version `6` is just ignored. Please add a test for that.

In should be `ipaddress.ip_address(host).version in [4, 6]`.

But I don't get how this expression can be `Any`.

Reply:
Note to reviewers. This exists in 3 other integrations:
```
homeassistant/components/dunehd/config_flow.py
20:        if ipaddress.ip_address(host).version == (4 or 6):

homeassistant/components/braviatv/config_flow.py
32:        if ipaddress.ip_address(host).version == (4 or 6):

homeassistant/components/vilfo/config_flow.py
35:        if ipaddress.ip_address(host).version == (4 or 6):

homeassistant/components/brother/config_flow.py
25:        if ipaddress.ip_address(host).version == (4 or 6):
```

Reply:
Typeshed has weird type annotations for `ipaddress` module:
https://github.com/python/typeshed/blob/master/stubs/ipaddress/ipaddress.pyi#L11

If mypy still complains about `Any` just cast it to `int`:
`cast(int, ipaddress.ip_address(host).version)`

Reply:
I'm a code owner of `dunehd` and `braviatv` so I'll fix this in those integrations.

Reply:
Awesome, thanks! Please fix `vilfo` as well 🙂

Reply:
Sure